chains/ethereum/config/src/types.rs

```diff
@@ -0,0 +1,440 @@
+    /// This call can be used to verify that the data you are pulling from is not tampered with.
+    #[derive(Encode, Decode)]
+    pub struct GetTransactionReceiptQuery {
+        tx_hash: TxHash,
+    }
+
+    impl EthQuery for GetTransactionReceiptQuery {
+        type Result = TransactionReceipt;
+    }
+
+    /// Executes a new message call immediately without creating a transaction on the block chain.
+    #[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+    pub struct CallContractQuery {
+        /// The address the transaction is sent from.
+        from: Option<Address>,
+        /// The address the transaction is directed to.
+        to: Address,
+        /// Integer of the value sent with this transaction.
+        value: U256,
+        /// Hash of the method signature and encoded parameters.
+        data: Bytes,
+        /// Call at block
+        block: BlockIdentifier,
+    }
+
+    /// The result of contract call execution
+    #[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+    pub enum CallResult {
+        /// Call executed succesfully
+        Success(Bytes),
+        /// Call reverted with message
+        Revert(Bytes),
+        /// Account doesn't exists
+        ContractNotFound,
+        /// Out of gas
+        OutOfGas,
+        /// Call is invalid
+        /// Ex: gas price > 64 bits
+        InvalidCall,
+    }
+
+    impl EthQuery for CallContractQuery {
+        type Result = CallResult;
+    }
+}
+
+pub mod config {
+    use parity_scale_codec::{Decode, Encode};
+
+    use super::{
+        primitives::{Address, Block, BlockIdentifier, TxHash, H256, U256},
+        queries::{
+            CallContractQuery, EthQuery, GetBalanceQuery, GetStorageAtQuery,
+            GetTransactionReceiptQuery,
+        },
+    };
+    use rosetta_core::traits::Config;
+
+    #[derive(Decode, Encode)]
+    pub enum Query {
+        /// Returns the balance of the account of given address.
+        GetBalance(GetBalanceQuery),
+        /// Returns the value from a storage position at a given address.
+        GetStorageAt(GetStorageAtQuery),
+        /// Returns the receipt of a transaction by transaction hash.
+        GetTransactionReceipt(GetTransactionReceiptQuery),
+        /// Executes a new message call immediately without creating a transaction on the block
+        /// chain.
+        CallContract(CallContractQuery),
+        /// Returns the account and storage values of the specified account including the
+        /// Merkle-proof. This call can be used to verify that the data you are pulling
+        /// from is not tampered with.
+        GetProof {
+            /// Address of the Account
+            address: Address,
+            /// an array of storage-keys that should be proofed and included
+            storage_keys: Vec<U256>,
+            /// State at the block
+            block: BlockIdentifier,
+        },
+    }
```

Comment:
This is the new definition of task for ethereum read-only tasks, this type is scale encoded/decoded by the connector.